static/app/views/organizationStats/teamInsights/teamStability.tsx

```diff
@@ -23,97 +21,87 @@ import {
-      periodSessions: null,
-    };
-  }
-
-  getEndpoints() {
-    const {organization, start, end, period, utc, projects} = this.props;
-
-    const projectsWithSessions = projects.filter(project => project.hasSessions);
-
-    if (projectsWithSessions.length === 0) {
-      return [];
-    }
+}
 
-    const datetime = {start, end, period, utc};
-    const commonQuery = {
-      environment: [],
-      project: projectsWithSessions.map(p => p.id),
-      field: 'sum(session)',
-      groupBy: ['session.status', 'project'],
-      interval: '1d',
-    };
-
-    const endpoints: ReturnType<AsyncComponent['getEndpoints']> = [
-      [
-        'periodSessions',
-        `/organizations/${organization.slug}/sessions/`,
-        {
-          query: {
-            ...commonQuery,
-            ...normalizeDateTimeParams(datetime),
-          },
+function TeamStability({
+  organization,
+  projects,
+  period,
+  start,
+  end,
+  utc,
+}: TeamStabilityProps) {
+  const projectsWithSessions = projects.filter(project => project.hasSessions);
+  const datetime = {start, end, period, utc};
+  const commonQuery = {
+    environment: [],
+    project: projectsWithSessions.map(p => p.id),
+    field: 'sum(session)',
+    groupBy: ['session.status', 'project'],
+    interval: '1d',
+  };
+
+  const {
+    data: periodSessions,
+    isLoading: isPeriodSessionsLoading,
+    isError: isPeriodSessionsError,
+    refetch: refetchPeriodSessions,
+  } = useApiQuery<SessionApiResponse>(
+    [
+      `/organizations/${organization.slug}/sessions/`,
+      {
+        query: {
+          ...commonQuery,
+          ...normalizeDateTimeParams(datetime),
         },
-      ],
-      [
-        'weekSessions',
-        `/organizations/${organization.slug}/sessions/`,
-        {
-          query: {
-            ...commonQuery,
-            statsPeriod: '7d',
-          },
+      },
+    ],
+    {staleTime: 5000}
+  );
+
+  const {
+    data: weekSessions,
+    isLoading: isWeekSessionsLoading,
+    isError: isWeekSessionsError,
+    refetch: refetchWeekSessions,
+  } = useApiQuery<SessionApiResponse>(
+    [
+      `/organizations/${organization.slug}/sessions/`,
+      {
+        query: {
+          ...commonQuery,
+          statsPeriod: '7d',
         },
-      ],
-    ];
+      },
+    ],
+    {staleTime: 5000}
+  );
 
-    return endpoints;
-  }
+  const isLoading = isPeriodSessionsLoading || isWeekSessionsLoading;
 
-  componentDidUpdate(prevProps: Props) {
-    const {projects, start, end, period, utc} = this.props;
-
-    if (
-      prevProps.start !== start ||
-      prevProps.end !== end ||
-      prevProps.period !== period ||
-      prevProps.utc !== utc ||
-      !isEqual(prevProps.projects, projects)
-    ) {
-      this.remountComponent();
-    }
+  if (isPeriodSessionsError || isWeekSessionsError) {
+    return (
+      <LoadingError
+        onRetry={async () =>
+          await Promise.all([refetchPeriodSessions(), refetchWeekSessions()])
+        }
```

Comment:
we don't need to wait for them since as soon as they're loading it kicks us back to the loading state.
```suggestion
        onRetry={() => {
          refetchPeriodSessions(), refetchWeekSessions()
        }}
```